cli/lib/kontena/cli/nodes/remove_command.rb

```diff
@@ -14,10 +14,8 @@ def execute
       nodes.each do |node_name|
         node = client(token).get("nodes/#{current_grid}/#{node_name}")
 
-        if node['has_token'] && node['connected']
-          warning "Node #{node['name']} is still connected using a node token, but will be force-disconnected"
-        elsif node['connected']
-          exit_with_error "Node #{node['name']} is still connected using a grid token. You must terminate the node before removing it."
+        if node['connected']
+          exit_with_error "Node #{node['name']} is still connected. You must terminate the node before removing it, for example: kontena cloud node rm"
```

Comment:
The `kontena cloud node rm` command will also remove the node from the grid, so the `You must terminate the node before removing it` doesn't make sense in that case.

Reply:
Something like this would be more useful, with explicit `kontena $plugin node terminate` hints and a general warning for offline nodes.

```ruby
provider = Array(node["labels"]).find{ |l| l.start_with?('provider=')}.to_s.split('=').last

if node['connected'] && provider
  exit_with_error "Node #{node['name']} is still connected. You should terminate the node instead: kontena #{provider} node terminate #{node['name']}"
elsif node['connected']
  exit_with_error "Node #{node['name']} is still connected. You must terminate the node before removing it"
else
  warning "Removing a node from the grid does not terminate the host machine from any cloud provider, so your cloud provider may continue to bill you for the machine until you terminate it"
end
```

The warning is to cover the edge cases of nodes that are just temporarily offline...

Reply:
Although maybe the warning should only be `if provider`... I suppose the main usecase for `kontena node rm` is self-hosted custom nodes, which should be provisioned without any `provider=...` labels, and warning about generic cloud providers is unnecessary.